api/src/DataProvider/TopBookItemDataProvider.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DataProvider;
+
+use ApiPlatform\Core\DataProvider\ItemDataProviderInterface;
+use ApiPlatform\Core\DataProvider\RestrictedDataProviderInterface;
+use ApiPlatform\Core\Exception\InvalidIdentifierException;
+use App\Entity\TopBook;
+
+final class TopBookItemDataProvider implements ItemDataProviderInterface, RestrictedDataProviderInterface
+{
+    private TopBookDataProvider $dataProvider;
+
+    public function __construct(TopBookDataProvider $dataProvider)
+    {
+        $this->dataProvider = $dataProvider;
+    }
+
+    public function supports(string $resourceClass, string $operationName = null, array $context = []): bool
+    {
+        return TopBook::class === $resourceClass;
+    }
+
+    /**
+     * @throws InvalidIdentifierException
+     */
+    public function getItem(string $resourceClass, $id, string $operationName = null, array $context = []): ?TopBook
+    {
+        if (!is_int($id)) {
+            throw new InvalidIdentifierException('Invalid id key type.');
+        }
+        $id = $this->checkId($id);
+        try {
+            $topBooks = $this->dataProvider->getTopBooks();
+        } catch (\Exception $e) {
+            throw new \RuntimeException(sprintf('Unable to retrieve top books from external source: %s', $e->getMessage()));
+        }
+
+        if (!array_key_exists($id, $topBooks)) {
+            throw new InvalidIdentifierException(sprintf('Top book ranked n°"%d" not found.', $id));
+        }
+
+        return $topBooks[$id];
+    }
+
+    /**
+     * @throws InvalidIdentifierException
+     */
+    private function checkId(int $id): int
```

Comment:
I think this test is useless: if the id exists in the getTopBooks array, it'll be returned. Otherwise, the getItem method must return null and it'll throw a 404 error